docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,169 @@
+In the example above, `hive_foo_counter` and `hive_foo_gauge` will only be logged if the value reported for them exceeds 10 minutes.
+
+#### additionalClusterDeploymentLabels
+Deprecate current implementation of hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels and change it to be reported as metricsConfig.metricsToReport.metrics[$name].additionalClusterDeploymentLabels. Its implementation will not change.
+In the example above, `hive_foo_counter` and `hive_foo_gauge` would report an additional label `prom_label_name`, its value corresponding to the value of `hive.openshift.io/cd-label-key` label on the corresponding ClusterDeployment.
+
+#### clusterDeploymentLabelSelector
+This would be a new feature, of type [LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), and we'd use the LabelSelector.MatchLabels and/or LabelSelector.MatchExpressions to match the conditions in order to decide if a metric should be reported.
+This encapsulates slightly more advanced filter logic over the existing ClusterDeployment labels. In the example above, `hive_foo_counter` and `hive_foo_gauge` metrics will only be reported for the ClusterDeployments that are labelled with aro-snowflake and not in limited support.
+
+### Implementation Details / Notes
+
+- All the options configured for a metric within metricsToReport, will work in tandem with each other. For ex, if all possible options are specified for a metric, then that metric will be reported with the additional labels as per additionalClusterDeploymentLabels, and will be reported only if it matches the labels and/or expressions as per clusterDeploymentLabelSelector and if the duration to be reported exceeds the minimumDuration
+- This design allows for grouping of metrics for which you would want to define particular customizations. Grouping allows hiveconfig to be parsable, but implementation of this model would require an efficient solution to validate there are no ambiguous entries for a metric, and storing it within the code for parsing purposes. We might end up enforcing only 1 entry for a metric in the HiveConfig.
+- We would be removing the fixed labels that can be propagated from the cluster deployment, i.e.`cluster_type`, `platform` and `version`. This means, if admins want these labels for the metrics, they will have to define them via  metricsConfig.metricsToReport[].additionalClusterDeploymentLabels. This would only be a breaking change if the admins suddenly cut over from deprecated/old way of metricsConfig to the new metricsToReport, sufficient documentation should help though.
+- We would be renaming `hive_cluster_deployment_install_failure_total` and `hive_cluster_deployment_install_success_total` to ensure it ends with `seconds`, as it is a duration-based metric.
```

Comment:
This is concerning, as it's an _actual_ breaking change. Everything else will be invisible to the user when we deploy the new build, giving them the power to choose when they cut over. But this will silently break existing graphs/alerts.

Much as I hate to do it, let's clone the existing metrics instead of outright replacing them. Then we can remove the old copies after the deprecation period is over. Glad it's only two :P

Reply:
I had not mentioned so explicitly, but it would be cloning anyways because old way and new way will exist side-by-side until the deprecation period is over. Also, remember, the new way makes all metrics to be optional - so the user will need to specify the metric in hiveConfig - they will notice it early on. Included a note.
(I'm also glad it's only 2. I think I added these metrics a while ago actually, and the only reason we missed it at that time is because our naming conventions for metrics are not documented - just somehow implemented in code. That's why I want to correct it by documenting it, so it is easy for future maintenance)